app/javascript/react/screens/App/Settings/SettingsActions.js

```diff
@@ -61,6 +66,36 @@ export const fetchConversionHostsAction = url => {
   return _getConversionHostsActionCreator(uri.toString());
 };
 
-export const showConversionHostWizard = () => dispatch => dispatch({ type: SHOW_V2V_CONVERSION_HOST_WIZARD });
+export const showConversionHostWizardAction = () => dispatch => dispatch({ type: SHOW_V2V_CONVERSION_HOST_WIZARD });
 
-export const hideConversionHostWizard = () => dispatch => dispatch({ type: HIDE_V2V_CONVERSION_HOST_WIZARD });
+export const hideConversionHostWizardAction = () => dispatch => dispatch({ type: HIDE_V2V_CONVERSION_HOST_WIZARD });
+
+export const conversionHostWizardExitedAction = () => dispatch => {
+  dispatch({ type: V2V_CONVERSION_HOST_WIZARD_EXITED });
+  // Dispatch reset for all the wizard step forms here
+  Object.values(stepIDs).forEach(formName => {
+    dispatch(reset(formName));
+  });
+};
+
+const _getProvidersActionCreator = url => dispatch =>
+  dispatch({
+    type: FETCH_V2V_PROVIDERS,
+    payload: API.get(url)
+  });
+
+export const fetchProvidersAction = url => {
```

Comment:
Food for thought...

There is already a system in place for fetching and maintaining a list of providers in the Overview screen.  Now that we need similar functionality here in Settings, we are duplicating that code.

When resources are shared across screens like this, it might make sense to have "application level" actions/reducers.  These could be made available to the screens/components that need them in the same way we're already doing it (map state to props in the container components)

A few benefits for this approach
1. Reduce code duplication
2. It would reduce the size of our reducers and make them easier to read.
3. Screen reducers would only need to concern themselves with the UI state and resource reducers would manage data

That being said, the work here is in line with the state management system that is already in place.  I just wanted to point this out

Reply:
@michaelkro I did a major refactor to create the new `providersReducer` and `targetResourcesReducer` based on this advice. I'm curious what you think of this implementation.